lib/datadog/tracing/contrib/aws/instrumentation.rb

```diff
@@ -60,6 +64,96 @@ def annotate!(span, context)
             span.set_tag(Tracing::Metadata::Ext::HTTP::TAG_STATUS_CODE, context.safely(:status_code))
           end
 
+          def add_service_specific_tags(span, aws_service, params)
+            case aws_service
+            when 'sqs'
+              add_sqs_tags(span, params)
+            when 'sns'
+              add_sns_tags(span, params)
+            when 'dynamodb'
+              add_dynamodb_tags(span, params)
+            when 'kinesis'
+              add_kinesis_tags(span, params)
+            when 'eventbridge'
+              add_eventbridge_tags(span, params)
+            when 'states'
+              add_states_tags(span, params)
+            when 's3'
+              add_s3_tags(span, params)
+            end
+          end
+
+          def add_sqs_tags(span, params)
+            queue_url = params.fetch(:queue_url, nil)
+            queue_name = params.fetch(:queue_name, nil)
+            if queue_url
+              # example queue_url: https://sqs.sa-east-1.amazonaws.com/12345678/MyQueueName
+              queue_name = queue_url.split('/')[-1]
+              aws_account = queue_url.split('/')[-2]
+              span.set_tag(Ext::TAG_AWS_ACCOUNT, aws_account)
+            end
+            span.set_tag(Ext::TAG_QUEUE_NAME, queue_name)
+          end
+
+          def add_sns_tags(span, params)
+            topic_arn = params.fetch(:topic_arn, nil)
+            topic_name = params.fetch(:name, nil)
+            if topic_arn
+              # example topic_arn: arn:aws:sns:us-west-2:123456789012:my-topic-name
+              topic_name = topic_arn.split(':')[-1]
+              aws_account = topic_arn.split(':')[-2]
+              span.set_tag(Ext::TAG_AWS_ACCOUNT, aws_account)
+            end
+            span.set_tag(Ext::TAG_TOPIC_NAME, topic_name)
+          end
+
+          def add_dynamodb_tags(span, params)
+            table_name = params.fetch(:table_name, '')
+            span.set_tag(Ext::TAG_TABLE_NAME, table_name)
+          end
+
+          def add_kinesis_tags(span, params)
+            stream_arn = params.fetch(:stream_arn, nil)
+            stream_name = params.fetch(:stream_name, '')
+            if stream_arn
+              # example stream_arn: arn:aws:kinesis:us-east-1:123456789012:stream/my-stream
+              stream_name = stream_arn.split('/')[-1] rescue ''
+              aws_account = stream_arn.split(':')[-2] rescue ''
```

Comment:
When do you expect these to fail? I ask because of the `rescue ''` here.

Reply:
I didn't have a specific case in mind where these would fail however I tried to write a test where I inputed an invalid stream_arn and the aws-sdk would fail the test.

Reply:
I think the rescues are not needed